compiler/rustc_hir_analysis/src/collect.rs

```diff
@@ -1025,7 +1025,15 @@ fn adt_def(tcx: TyCtxt<'_>, def_id: LocalDefId) -> ty::AdtDef<'_> {
 
     let is_anonymous = item.ident.name == kw::Empty;
     let repr = if is_anonymous {
-        tcx.adt_def(tcx.local_parent(def_id)).repr()
+        let parent = tcx.local_parent(def_id);
+        if let Node::Item(item) = tcx.hir_node_by_def_id(parent)
+            && item.is_struct_or_union()
+        {
+            tcx.adt_def(tcx.local_parent(def_id)).repr()
```

Comment:
```suggestion
            tcx.adt_def(parent).repr()
```

Oh, did I miss that? After that we should be good to go for good ^^'!